src/control/server/ctl_storage_rpc.go

```diff
@@ -852,6 +855,12 @@ func formatScm(ctx context.Context, req formatScmReq, resp *ctlpb.StorageFormatR
 		}
 	}
 
+	if req.replace && len(needFormat) == 0 {
+		// Only valid if at least one engine requires format.
+		return nil, nil, errors.New("format replace option only valid if at " +
+			"least one engine requires format but no engines need format")
```

Comment:
This message is not very clear to me, do you mean:
```suggestion
			"least one engine requires format but no other engines need format")
```
If yes, the test should test `len(needFormat) != 1`